src/diff_edit.rs

```diff
@@ -96,6 +135,122 @@ fn set_readonly_recursively(path: &Path) -> Result<(), std::io::Error> {
     }
 }
 
+pub fn run_mergetool(
+    _ui: &mut Ui,
+    _settings: &UserSettings,
+    tree: &Tree,
+    repo_path: &RepoPath,
+) -> Result<TreeId, DiffEditError> {
+    let conflict_id = match tree.path_value(repo_path) {
+        Some(TreeValue::Conflict(id)) => id,
+        Some(_) => return Err(DiffEditError::NotAConflictError(repo_path.clone())),
+        None => return Err(DiffEditError::PathNotFoundError(repo_path.clone())),
+    };
+    let conflict = tree.store().read_conflict(repo_path, &conflict_id)?;
+    let mut materialized_conflict: Vec<u8> = vec![];
+    materialize_conflict(
+        tree.store(),
+        repo_path,
+        &conflict,
+        &mut materialized_conflict,
+    )
+    .unwrap_or_else(|_| materialized_conflict.clear());
+    // For normal files, this is the default user-facing represenation of conflicts
+    // as a text file with conflict markers.
+    let materialized_conflict = materialized_conflict;
+
+    let content = match extract_file_conflict_data(tree.store(), repo_path, &conflict) {
+        Some(c) => c,
+        _ => {
+            // In this case, `materialized_conflict` is a short summary of what kind of
+            // files are involved in the conflict.
+            let summary = match std::str::from_utf8(materialized_conflict.as_slice()) {
+                Ok(s) if !materialized_conflict.is_empty() => s.to_string(),
```

Comment:
As noted above (though I may be mistaken), we should never have failed to create a description since it was written to memory, so drop the `if !materialized_conflict.is_empty()`?